### PR TITLE
LP-1880 Implementation to restrict users to two sets of referral emails

### PR DESCRIPTION
**Pull request by zahid-arbisoft**
### Description

[LP-1880](https://philanthropyu.atlassian.net/browse/LP-1880) Implementation to restrict users to two sets of referral emails

### Notes

Unit test included

### CODE REVIEW

File: openedx/features/smart_referral/helpers.py [hunk 1]
```diff
@@ -57,3 +59,23 @@ def get_email_domain(email):
...
+
+    filtered_contacts = []
+    for contact in contacts:
+        qs_referral = SmartReferral.objects.filter(contact_email=contact.get('contact_email'))
+        qs_count = qs_referral.count()
+
+        # add contact to list if it has never been referred before or if it was referred
+        # only once by any other user (not by current user)
+        is_referred_before = qs_count > 1 or (qs_count == 1 and qs_referral.first().user == user)
+        filtered_contacts.append(contact) if not is_referred_before else None
```
Comment (ahmed-faraz46) [inline openedx/features/smart_referral/helpers.py:79]: how about?
```
if qs_count == 0 or qs_reffered.first().user != user:
    filtered_contacts.append(contact)
```

Now if we give qs_count and qa_reffered some more meaningful names, we don't even need to add the comment to explain the logic.

Comment (zahid-arbisoft) [reply openedx/features/smart_referral/helpers.py:79]: This condition will not work, we need to cater following scenarios:

1. If `qs_count` >=2, `is_referred_before` must return True
2. If `qs_count` is exactly 1 and `first().user` is same as current user then `is_referred_before ` must return True
3. If `qs_count` == 0, `is_referred_before` must return Flase

Comment (ahmed-faraz46) [reply openedx/features/smart_referral/helpers.py:79]: Just forget about is_reffered_before, consider actual criteria you have mentioned in the code and lemme know if any case is not tackled by this condition.

File: openedx/features/smart_referral/helpers.py [hunk 2]
```diff
@@ -57,3 +59,23 @@ def get_email_domain(email):
     Return email domain name e.g "google.com", "yahoo.com" and "outlook.com" etc by splitting email after "@".
     """
     return email.split('@')[1] if email else None
+
+
+def filter_referred_contacts(contacts, user):
```
Comment (monofal) [inline openedx/features/smart_referral/helpers.py:64]: In the AC it says 

- Filter out contact if it has already registered on our platform

I can't seem to find that filter here. How we are handling that logic?

Comment (zahid-arbisoft) [reply openedx/features/smart_referral/helpers.py:64]: We are already showing only non-platform contact on the widget. So, I have added my filter on **non-platform contacts** only. See this [line](https://github.com/philanthropy-u/edx-platform/pull/1644/files#diff-18e5c3d4ca9be3c6408a21ce14e264c8R40)

**Pull request merged**